packages/adonis/src/AdonisAdapter.ts

```diff
@@ -0,0 +1,138 @@
+import { readFileSync } from 'fs';
+import ejs from 'ejs';
+import mime from 'mime';
+
+import {
+  AppControllerRoute,
+  AppViewRoute,
+  BullBoardQueues,
+  ControllerHandlerReturnType,
+  HTTPMethod,
+  IServerAdapter,
+  UIConfig,
+} from '@bull-board/api/dist/typings/app.js';
+
+import { MiddlewareFn, OneOrMore, ParsedNamedMiddleware } from '@adonisjs/core/types/http';
+import { HttpRouterService } from '@adonisjs/core/types';
+import type { HttpContext } from '@adonisjs/core/http';
+
+export class AdonisAdapter implements IServerAdapter {
+  protected basePath = '';
+  protected bullBoardQueues: BullBoardQueues | undefined;
+  protected errorHandler: ((error: Error) => ControllerHandlerReturnType) | undefined;
+  protected uiConfig: UIConfig = {};
+  private viewPath: string | undefined;
+  private statics?: { path: string; route: string };
+  private middlewares!: OneOrMore<MiddlewareFn | ParsedNamedMiddleware>;
+  private router: HttpRouterService;
+
+  constructor(router: HttpRouterService) {
+    this.router = router;
+  }
+
+  public use(middlewares: OneOrMore<MiddlewareFn | ParsedNamedMiddleware>): this {
+    this.middlewares = middlewares;
+    return this;
+  }
+
+  public setBasePath(path: string): this {
+    this.basePath = path;
+    return this;
+  }
+
+  public setStaticPath(staticsRoute: string, staticsPath: string): this {
+    this.statics = { route: staticsRoute, path: staticsPath };
+    return this;
+  }
+
+  public setViewsPath(viewPath: string): this {
+    this.viewPath = viewPath;
+    return this;
+  }
+
+  public setQueues(bullBoardQueues: BullBoardQueues): this {
+    this.bullBoardQueues = bullBoardQueues;
+    return this;
+  }
+
+  public setUIConfig(config: UIConfig = {}): this {
+    this.uiConfig = config;
+    return this;
+  }
+
+  public setErrorHandler(handler: (error: Error) => ControllerHandlerReturnType) {
+    this.errorHandler = handler;
```

Comment:
I don't see that there is any usage of this errorHandler from Adonis part... did you missed it?